src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEReader.cs

```diff
@@ -385,6 +385,12 @@ private AbstractMemoryBlock GetPESectionBlock(int index)
                 Interlocked.CompareExchange(ref _lazyPESectionBlocks, new AbstractMemoryBlock[PEHeaders.SectionHeaders.Length], null);
             }
 
+            AbstractMemoryBlock? existingBlock = Volatile.Read(ref _lazyPESectionBlocks[index]);
```

Comment:
Could this new code go under an `else` for the `if` statement above?

Reply:
I considered it, but there is a rare case where this code races with another thread and that thread is able to fill the cache.  By leaving this out of the else we give ourselves that small window to get a cache hit.